scripts/push_php_sdk.py

```diff
@@ -0,0 +1,157 @@
+import os
+from os import path
+from datetime import datetime
+
+from git_client import GitClient, GitException
+from fs_client import FsClient
+import utils
+
+logger = utils.get_logger()
+
+class PushPhpSdkPipeline:
+  def __init__(self, git_client, fs_client, criteo_service, api_version):
+    self.criteo_service = criteo_service
+    self.api_version = api_version
+    self.generator_version = 0
+    self.__init_environment_variables()
+
+    self.fs = fs_client
+    self.fs.change_dir(self.generator_repo_dir)
+
+    self.git = git_client
+
+    logger.info(f'Setup Git with user {self.github_actor}.')
+    self.git.setup(self.github_actor)
+  
+  def clone_repo(self):
+    self.fs.change_dir(self.sdk_repo_dir)
+
+    organization_name = 'criteo'
+    repository_name = f'criteo-api-{self.criteo_service}-php-sdk'
+  
+    logger.info(f'Cloning repository {organization_name}/{repository_name}...')
+
+    self.git.clone(organization_name, repository_name)
+
+    self.sdk_repo_dir = path.join(self.sdk_repo_dir, repository_name)
+
+  def checkout(self):
+    self.fs.change_dir(self.sdk_repo_dir)
+  
+    branch_name = self.api_version
+    if self.criteo_service != 'preview':
+      branch_name = branch_name.replace('-', '.')
+
+    logger.info(f'Checkout branch {branch_name}.')
+
+    self.git.checkout(branch_name)
+
+  def update_sources(self):
+    logger.info('Copying new sources to repository...')
+
+    el_to_update = ['docs', 'examples', 'lib', 'test', '.gitignore', '.php_cs', 'README.md', 'composer.json', 'composer.lock', 'phpunit.xml.dist']
```

Comment:
Why not copying all the folder content and adding a gitignore to discard what's not interesting?